tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1091,11 +1091,20 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITInlining(FunctionID callerId, Function
 //
 // InitializeProfiler method
 //
-void CorProfiler::InitializeProfiler(CallTargetDefinition* items, int size)
+void CorProfiler::InitializeProfiler(WCHAR* id, CallTargetDefinition* items, int size)
 {
     auto _ = trace::Stats::Instance()->InitializeProfilerMeasure();
 
-    Logger::Info("InitializeProfiler: received from managed side: ", size, " integrations.");
+    WSTRING definitionsId = WSTRING(id);
+    Logger::Info("InitializeProfiler: received id: ", definitionsId, " from managed side with ", size,
+                 " integrations.");
+
+    if (definitions_ids_.find(definitionsId) != definitions_ids_.end())
```

Comment:
shouldn't the check in the set be done covered by a lock ?

Reply:
you're right, fixing it.

Reply:
done in 3925992